src/model_manager/model.cpp

```diff
@@ -22,6 +22,8 @@ void Model::LoadModelDetails(const nlohmann::json& model_json) {
     model_details_.provider_name =
         model_json.contains("provider") ? model_json.at("provider").get<std::string>() : std::get<1>(query_result);
     auto secret_name = "__default_" + model_details_.provider_name;
+    if (model_details_.provider_name == "azure")
```

Comment:
We should not have an azure literal thrown in there, these constants should all be in one place. No?